tests/test_correctbiasutils.py

```diff
@@ -60,3 +62,17 @@
 ])
 def testDivideGenomeRand(regions, baseBinSize, genomeBinSize, result):
 	assert divideGenome(regions, baseBinSize, genomeBinSize) == result
+
+@pytest.mark.parametrize("inputArray,numBins,result", [
+	([1, 2, 3, 4, 5, 6, 7, 8, 9], 1, [[1, 2, 3, 4, 5, 6, 7, 8, 9]]),
+	([1, 2, 3, 4, 5, 6, 7, 8, 9], 2, [[1, 2, 3, 4], [5, 6, 7, 8, 9]]),
+	([1, 2, 3, 4, 5, 6, 7, 8, 9], 3, [[1, 2, 3], [4, 5, 6], [7, 8, 9]]),
+	([1, 2, 3, 4, 5, 6, 7, 8, 9], 4, [[1, 2], [3, 4], [5, 6], [7, 8, 9]]),
+	([1, 2, 3, 4, 5, 6, 7, 8, 9], 5, [[1], [2], [3], [4], [5, 6, 7, 8, 9]]),
```

Comment:
@Young-Sook the results for 5 and 6 bins here are kind of funky. For 5 bins I think 
```python
[[1, 2], [3, 4], [5, 6], [7, 8], [9]]
```

would be a nicer packing. and for 6
```python
[[1, 2], [3, 4], [5, 6], [7], [8], [9]]
```
might also be nicer. 

I don't necessarily think these bin packings are wrong, based on the criteria, just surprising and a little funky.

Reply:
Hm, what about these bin packings?

5:
```python
[[1], [2, 3], [4, 5], [6, 7], [8, 9]]
```

6:
```python
[[1], [2], [3], [4, 5], [6, 7], [8, 9]]
```

I think I can come up with a reasonably efficient algorithm for dividing bins up so that no bin is ever more than one larger than any other bin, and all the larger bins would be at the end.

Reply:
I think it is great itself, but I wonder how we should edit this line:
https://github.com/ReddyLab/CRADLE/blob/main/CRADLE/CorrectBias/calculateOnebp.pyx#L42

That line assumes only the last bin has different size, so we could use the uniform bin size (vari.BINSIZE) in other bins. However, if we adapt that array-split method, we might need to add more lines to check how many bins have different bin size from vari.BINSIZE? If this problem is resolved, I think that array splitting method would be great!

Reply:
Correctbias wouldn't be affected by this change at all right now because it doesn't use the "common" functions that use arraySplit. So this won't be an issue until I start updating `CorrectBias/` to use `utils.getCandidateTrainingSet`.

Reply:
But to answer your question, major surgery would be required on the correctbias code to incorporate this new binning. I think it's good surgery that will improve things, but it will be a lot of changes.